wgpu-native/src/device.rs

```diff
@@ -29,15 +32,18 @@ pub fn wgpu_create_surface(raw_handle: raw_window_handle::RawWindowHandle) -> id
                 .create_surface_from_uiview(h.ui_view, cfg!(debug_assertions)),
         },
         #[cfg(target_os = "macos")]
-        Rwh::MacOS(h) => core::instance::Surface {
-            #[cfg(feature = "vulkan-portability")]
-            vulkan: instance
-                .vulkan
-                .as_ref()
-                .map(|inst| inst.create_surface_from_ns_view(h.ns_view)),
-            metal: instance
-                .metal
-                .create_surface_from_nsview(h.ns_view, cfg!(debug_assertions)),
+        Rwh::MacOS(h) => {
+            let ns_view = get_default_nsview_if_null(h.ns_view, h.ns_window);
```

Comment:
perhaps, we could just write the code in place here? i.e.
```rust
let ns_view = if h.ns_view.is_null() {
  let ns_window = h.ns_window as *mut Object;
  unsafe { msg_send![ns_window, contentView] };
} else {
  h.ns_view
};
```

Reply:
Yeah that does look better, made the change :)